plugins/imgur.py

```diff
@@ -128,7 +129,15 @@ def export_submission(self,
             if is_album:
                 results['link_display'] = '[Imgur Album](https://imgur.com/a/%s)  \n' % album['id']
             else:
-                results['link_display'] = '[Imgur](%s)  \n' % images[0]['link'].replace('http', 'https')
+                
+                picture_url = images[0]['link'].replace('http', 'https')
+                r = requests.head(picture_url)
+                mime_text = r.headers.get('Content-Type')
+                mime = mimeparse.parse_mime_type(mime_text)
+                if mime[1] == 'gif':
+                    picture_url = re.sub('\.gif|jpg|png', '.gifv', picture_url)
```

Comment:
Alright, that should do it then. Thanks a lot. I was going to refine the regular expression, but you did it before I could. And thanks for the explanation on raw strings. I know C# and Python have them, but I'm just not used to them as I'm mainly a Java programmer.